src/lib.rs

```diff
@@ -349,7 +349,7 @@ macro_rules! init_default_environment {
             let sctk_primary_selection_manager = $crate::primary_selection::PrimarySelectionHandler::init(&mut sctk_seats);
 
             let display = $crate::reexports::client::Proxy::clone(&$display);
-            let env = $crate::environment::Environment::init(&display.attach($queue.token()), $env_name {
+            let env = $crate::environment::Environment::new_pending(&display.attach($queue.token()), $env_name {
```

Comment:
I think we can replace this call by a call to `::new()` and drop the two roundtrips in the macro.

Reply:
We need to setup primary selection manager, and since it's not a function we can't use `?` syntax here so I've done it that way to avoid manually doing `if let`.

Reply:
I think I'd prefer it with the `if let`, intent is clearer that way imo. That or a `.map()` on the result, works as well.

Reply:
I've updated PR.